### ISSUE TITLE
return first batch twice

### ISSUE THREAD

**Issue by shesung** [NVIDIA/DALI#153]
I install DALI 0.2.0 from pip. I write some code similar to https://github.com/NVIDIA/DALI/blob/master/docs/examples/external_input.ipynb . 

However, I get very strange behavior. The first batch will be return twice ! And (n-1) th data will be returned when n th input batch is feeded. 

The code is:
```
import os
import time
import numpy as np

from nvidia.dali.pipeline import Pipeline
import nvidia.dali.ops as ops
import nvidia.dali.types as types

batch_size = 4

images_dir = "./images/"
fns = os.listdir(images_dir)

# read data
datas = []
labels = []
for i in range(0,12):
    f = open(os.path.join(images_dir, fns[i%len(fns)]))
    d = np.frombuffer(f.read(), dtype = np.uint8)
    datas.append(d)
    labels.append(np.array([i], dtype = np.uint8))

# define pipeline
class ExternalSourcePipeline(Pipeline):                   
    def __init__(self, batch_size, num_threads, device_id):
        super(ExternalSourcePipeline, self).__init__(batch_size,
                                      num_threads,
                                      device_id,
                                      seed=12)
        self.input = ops.ExternalSource()
        self.input_label = ops.ExternalSource()
        self.decode = ops.nvJPEGDecoder(device = "mixed", output_type = types.RGB)

    def define_graph(self):                                                                
        self.jpegs = self.input()
        self.labels = self.input_label()
        output = self.decode(self.jpegs)                                                
        return (output, self.labels)

pipe = ExternalSourcePipeline(batch_size=4, num_threads=2, device_id = 0)
pipe.build()

for i in range(0,12,4):
    print("batch %d:"%(i/4)),
    pipe.feed_input(pipe.jpegs, datas[i:i+4])
    pipe.feed_input(pipe.labels, labels[i:i+4])
    pipe_out = pipe.run()
    batch_cpu = pipe_out[0].asCPU()
    labels_cpu = pipe_out[1]
    for x in range(4):
        print(labels_cpu.at(x)),
    print
    time.sleep(1)
```

the output i got:

```
batch 0: [0] [1] [2] [3]
batch 1: [0] [1] [2] [3]
batch 2: [4] [5] [6] [7]
```

If I remove this line:
`    time.sleep(1)`

It will run into the following error:

```
RuntimeError: Critical error in pipeline: Error in thread 1: [/opt/dali/dali/pipeline/operators/decoder/nvjpeg_decoder.h:327] NVJPEG error "3"
Current pipeline object is no longer valid.
```

Any advices? 
Thanks!

**Comment by JanuszL** [comment]
Tracked as DALI-228

**Comment by JanuszL** [comment]
Hi,
It was fixed in #106 and will be available with 0.3 release.

**Comment by shesung** [comment]
Thanks! problem solved.
